### DISCUSSION TITLE
How To get an expanded version of a JSON schema that eliminates $ref and allOf references

### DISCUSSION THREAD

**Discussion by adarshasp87** [networknt/json-schema-validator#1125] [General]
How To get an expanded version of a JSON schema that eliminates $ref and allOf references, programmatically resolve these constructs into a single schema definition?

**Comment by stevehu** [comment]
I think you want to resolve all the references locally. We don't have any example to do so with JSON schemas but there is an example to merge and resolve all references for OpenAPI specifications. Here is the repo and you can use it as a reference to build your own. 

https://github.com/networknt/openapi-bundler

**Comment by mpenet** [comment]
io.swagger.v3.parser.OpenAPIV3Parser is worth mentioning also, it does that without too much fuss, see ParseOptions/.setResolveFully. 

```java
import io.swagger.v3.parser.OpenAPIV3Parser;
import io.swagger.v3.parser.core.models.ParseOptions;
import io.swagger.v3.parser.core.models.SwaggerParseResult;

public class OpenAPIParserExample {
    public static void main(String[] args) {
        OpenAPIV3Parser parser = new OpenAPIV3Parser();
        ParseOptions options = new ParseOptions();
        options.setResolveFully(true);
        options.setValidateExternalRefs(true);

        SwaggerParseResult result = parser.readContents("{}", null, options);
    }
}
```
